java-frontend/src/main/java/org/sonar/java/se/CheckerContext.java

```diff
@@ -45,4 +45,12 @@
   ExplodedGraph.Node getNode();
 
   ConstraintManager getConstraintManager();
+
+  AlwaysTrueOrFalseExpressions alwaysTrueOrFalseExpressions();
+
+  interface AlwaysTrueOrFalseExpressions {
```

Comment:
I am struggling to see the point of this interface. I would rely directly on the Object {{AlwaysTrueOrFalseExpressionCollector}} that would simplify things.